src/libraries/System.Private.CoreLib/src/System/MissingMemberException.cs

```diff
@@ -71,5 +74,184 @@ public override string Message
         protected string? ClassName;
         protected string? MemberName;
         protected byte[]? Signature;
+
+        internal static string FormatSignature(byte[]? signature)
+        {
+            if (signature == null || signature.Length == 0)
+            {
+                return "Unknown signature";
+            }
+
+            ValueStringBuilder stringBuilder = new ValueStringBuilder(stackalloc char[256]);
+
+            try
+            {
+                ReadOnlySpan<byte> sig = signature;
+
+                int cconv = BinaryPrimitives.ReadInt32LittleEndian(sig);
+                sig = sig.Slice(sizeof(int));
+
+                if (cconv == (int)CorCallingConvention.IMAGE_CEE_CS_CALLCONV_FIELD)
+                {
+                    UnparseType(sig, ref stringBuilder);
+                }
+                else
+                {
+                    int nargs = BinaryPrimitives.ReadInt32LittleEndian(sig);
+                    sig = sig.Slice(sizeof(int));
+
+                    // return type
+                    sig = sig.Slice(UnparseType(sig, ref stringBuilder));
+                    stringBuilder.Append('(');
+
+                    for (int i = 0; i < nargs; i++)
+                    {
+                        sig = sig.Slice(UnparseType(sig, ref stringBuilder));
+                        if (i != nargs - 1)
+                        {
+                            stringBuilder.Append(", ");
+                        }
+                    }
+
+                    stringBuilder.Append(')');
+                }
+
+                return stringBuilder.ToString();
+            }
+            catch (Exception ex) when (ex is IndexOutOfRangeException or ArgumentOutOfRangeException)
+            {
+                stringBuilder.Dispose();
+                throw new ArgumentException(SR.Argument_BadSigFormat, nameof(signature));
+            }
+        }
+
+        private static int UnparseType(ReadOnlySpan<byte> sig, ref ValueStringBuilder stringBuilder)
```

Comment:
I haven't seen method using `ref ReadOnlySpan` to avoid the requirement of bytesRead. What's the suggested pattern here?

Reply:
Returning `int` with number of bytes parsed is fine.